src/main/java/kostka/moviecatalog/entity/User.java

```diff
@@ -0,0 +1,64 @@
+package kostka.moviecatalog.entity;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotEmpty;
+import java.io.Serializable;
+import java.util.Set;
+
+@Entity
+@SequenceGenerator(name = "seq", allocationSize = 1)
+@Table(name = "user_account")
+public class User implements Serializable {
+
+    private static final long serialVersionUID = -3350439623038263787L;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seq")
+    @Column(name = "user_id")
+    private Long userId;
+
+    @NotEmpty(message = "Name cannot be empty")
+    @Column(name = "username", nullable = false)
+    private String username;
+
+    @Column(name = "password", nullable = false)
+    private String password;
+
+    @ManyToMany(cascade = CascadeType.DETACH, fetch = FetchType.EAGER)
```

Comment:
Potřebuješ EAGER?
Přistupuješ k rolím vždy, když načteš usera?

Reply:
Aááá, tento koment možná beru zpět. Use se ukládá to té Spring Security entity, že jo.
Takže je možné, že getRoles() se někde volá i mimo otevřenou DB session. Pak je EAGER potřeba.

No, člověk se pořád učí. Blbuvzdorné by bylo asi do Security, ukládat nějaké DTO. Možná se na to zkusím podívat i na luštěninách.

Reply:
presne tak, pouzil jsem to z lustenin :)